weaviate/batch/crud_batch.py

```diff
@@ -257,29 +271,29 @@ def __init__(self, connection: Connection):
         self._objects_batch = ObjectsBatchRequest()
         self._reference_batch = ReferenceBatchRequest()
         # do not keep too many past values, so it is a better estimation of the throughput is computed for 1 second
-        self._objects_throughput_frame = deque(maxlen=5)
-        self._references_throughput_frame = deque(maxlen=5)
-        self._future_pool = []
-        self._reference_batch_queue = []
+        self._objects_throughput_frame: Deque[float] = deque(maxlen=5)
+        self._references_throughput_frame: Deque[float] = deque(maxlen=5)
+        self._future_pool: List[Future[Tuple[Response | None, int]]] = []
+        self._reference_batch_queue: List[ReferenceBatchRequest] = []
         self._callback_lock = threading.Lock()
 
         # user configurable, need to be public should implement a setter/getter
         self._callback: Optional[Callable[[BatchResponse], None]] = check_batch_result
         self._weaviate_error_retry: Optional[WeaviateErrorRetryConf] = None
         self._batch_size: Optional[int] = 50
-        self._creation_time = min(self._connection.timeout_config[1] / 10, 2)
+        self._creation_time = cast(Real, min(self._connection.timeout_config[1] / 10, 2))
         self._timeout_retries = 3
         self._connection_error_retries = 3
         self._batching_type: Optional[str] = "dynamic"
         self._recommended_num_objects = self._batch_size
         self._recommended_num_references = self._batch_size
 
         self._num_workers = 1
-        self._consistency_level = None
+        self._consistency_level: Optional[Literal["ALL", "ONE", "QUORUM"]] = None
```

Comment:
shouldn't we rather have the enum here?

Reply:
In all these changes, I tried to minimise the number of implementation changes by just making type changes but I agree that this should be the enum so will change the implementation elsewhere to align